packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/ListView/index.js

```diff
@@ -72,129 +65,95 @@ const ListView = () => {
   const webhooksToDeleteLength = webhooksToDelete.length;
   const getWebhookIndex = (id) => webhooks.findIndex((webhook) => webhook.id === id);
 
-  useEffect(() => {
-    isMounted.current = true;
-
-    return () => {
-      isMounted.current = false;
-    };
-  }, []);
-
-  /**
-   * TODO: refactor this, but actually refactor
-   * the whole component. Needs some love.
-   */
-  useEffect(() => {
-    const fetchWebHooks = async () => {
-      try {
-        const {
-          data: { data },
-        } = await get('/admin/webhooks');
-
-        if (isMounted.current) {
-          dispatch({
-            type: 'GET_DATA_SUCCEEDED',
-            data,
-          });
-          notifyStatus('webhooks have been loaded');
-        }
-      } catch (err) {
-        console.log(err);
-
-        if (isMounted.current) {
-          if (err.code !== 20) {
-            toggleNotification({
-              type: 'warning',
-              message: { id: 'notification.error' },
-            });
-          }
-          dispatch({
-            type: 'TOGGLE_LOADING',
-          });
-        }
-      }
-    };
+  const fetchWebHooks = async () => {
+    try {
+      const {
+        data: { data },
+      } = await get('/admin/webhooks');
 
-    if (canRead) {
-      fetchWebHooks();
-    }
-  }, [canRead, get, notifyStatus, toggleNotification]);
+      setWebhooks(data);
+      notifyStatus('webhooks have been loaded');
 
-  const handleToggleModal = () => {
-    setShowModal((prev) => !prev);
-  };
+      return data;
+    } catch (err) {
+      toggleNotification({
+        type: 'warning',
+        message: { id: 'notification.error' },
+      });
 
-  const handleConfirmDelete = () => {
-    if (webhookToDelete) {
-      handleConfirmDeleteOne();
-    } else {
-      handleConfirmDeleteAll();
+      return null;
     }
   };
+  const QUERY_KEY = 'webhooks';
+  const { isLoading: webhooksLoading } = useQuery(QUERY_KEY, fetchWebHooks);
+  const loading = RBACLoading || webhooksLoading;
 
-  const handleConfirmDeleteOne = async () => {
+  const { mutateAsync: handleConfirmDeleteOne } = useMutation(async () => {
```

Comment:
I would not destructure this and call it e.g. `webhookDeleteMutation` - it allows you to reference different methods later on, such as `webhookDeleteMutation.mutateAsync()`

I think I would also like if all mutations (delete ...) could be handled by one mutation, because only 1 can run in parallel and react-query then takes care to cancel the requests properly. ...